src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,101 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private int size;
+    private int capacity;
+    private int threshold;
+    private Node<K, V>[] table;
 
     @Override
     public void put(K key, V value) {
-
+        if (size == 0) {
+            initializeTable();
+        }
+        if (getNode(key) != null) {
+            getNode(key).value = value;
+            return;
+        }
```

Comment:
I guess it's a redundant logic here, because every time you trigger this logic, that it's not ok, it's enough to update logic from 28-31th, so in case key exists change its value and return; from the method